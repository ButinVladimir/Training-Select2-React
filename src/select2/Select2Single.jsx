import React from 'react';
import PropTypes from 'prop-types';
import Select2Main from './Select2Main';
import DefaultSelectedText from './components/selected-text/DefaultSelectedText';
import DefaultPopupMenu from './components/popup-menu/DefaultPopupMenu';
import DefaultInputTextField from './components/input-text-field/DefaultInputTextField';
import DefaultItemsList from './components/items-list/DefaultItemsList';
import DefaultListItemSingle from './components/items-list/DefaultListItemSingle';
import defaultPrepareSelectedText from './helpers/default-prepare-selected-text';
import performSelectionSingle from './helpers/perform-selection-single';

export default function Select2Single(props) {
  return (
    <Select2Main
      {...props}
      performSelection={performSelectionSingle}
    />
  );
}

Select2Single.propTypes = {
  onSelect: PropTypes.func.isRequired,
  getData: PropTypes.func.isRequired,
  prepareSelectedText: PropTypes.func,
  SelectedText: PropTypes.func,
  PopupMenu: PropTypes.func,
  InputTextField: PropTypes.func,
  ItemsList: PropTypes.func,
  ListItem: PropTypes.func,
};

Select2Single.defaultProps = {
  prepareSelectedText: defaultPrepareSelectedText,
  SelectedText: DefaultSelectedText,
  PopupMenu: DefaultPopupMenu,
  InputTextField: DefaultInputTextField,
  ItemsList: DefaultItemsList,
  ListItem: DefaultListItemSingle,
};
