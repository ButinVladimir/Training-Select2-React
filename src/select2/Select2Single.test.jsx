import React from 'react';
import renderer from 'react-test-renderer';
import Select2Single from './Select2Single';
import getData from '../get-data-local';

describe('Select2Single', () => {
  test('renders without crashing', () => {
    const component = renderer.create(
      <Select2Single
        onSelect={() => {}}
        getData={getData}
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
