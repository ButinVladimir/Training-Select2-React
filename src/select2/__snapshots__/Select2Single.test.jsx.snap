// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select2Single renders without crashing 1`] = `
<div
  className="select2-container"
>
  <input
    className="select2-selected-text"
    onClick={[Function]}
    readOnly={true}
    type="text"
    value="Click to select"
  />
</div>
`;
